### PR TITLE
Fix support for lcobucci/jwt v3.4 and 4.0

### LINKED ISSUE #793: php8 compatibility
After updating to php 8 I got that strange error in tests:
`InvalidArgumentException: It was not possible to parse your key, reason: 
/builds/riverwaysoft/rebelbee/phonedo_project/phonedo/vendor/lcobucci/jwt/src/Signer/OpenSSL.php:90
/builds/riverwaysoft/rebelbee/phonedo_project/phonedo/vendor/lcobucci/jwt/src/Signer/OpenSSL.php:47
/builds/riverwaysoft/rebelbee/phonedo_project/phonedo/vendor/lcobucci/jwt/src/Signer/OpenSSL.php:21
/builds/riverwaysoft/rebelbee/phonedo_project/phonedo/vendor/lcobucci/jwt/src/Signer/BaseSigner.php:36
/builds/riverwaysoft/rebelbee/phonedo_project/phonedo/vendor/lcobucci/jwt/src/Builder.php:470
/builds/riverwaysoft/rebelbee/phonedo_project/phonedo/vendor/lcobucci/jwt/src/Builder.php:450
/builds/riverwaysoft/rebelbee/phonedo_project/phonedo/vendor/lexik/jwt-authentication-bundle/Services/JWSProvider/LcobucciJWSProvider.php:99`
Reason of error is empty.

Problem pointed here:
Снимок экрана 2020-11-23 в 00 05 05

### LINKED ISSUE #795: Compatibility with lcobucci/jwt v3.4
Today, new versions for `lcobucci/jwt` were released, v3.4 and v4. In v3.4, a deprecation notice has been added, such that `\Lcobucci\JWT\Builder::convertToDate` should no longer get called with an integer (see https://github.com/lcobucci/jwt/blob/3.4/src/Builder.php#L151). This should be fixed in `\Lexik\Bundle\JWTAuthenticationBundle\Services\JWSProvider\LcobucciJWSProvider::create`

### PR DESCRIPTION
**Pull request by chalasr**
Fixes #795 
Fixes #793

### CODE REVIEW

**Comment by chalasr** [comment]
I'm going to merge this one in order to fix some openssl-related issues on PHP 8. 
If you see something wrong here, please tell me.

**Comment by lcobucci** [comment]
@chalasr have you considered the usage of the configuration object? That removes the for all the checks you're using here. You'd need to change the requirements to `^3.4 || ^4.0` (which should be fine)

**Comment by lcobucci** [comment]
Here's an example https://github.com/thephpleague/oauth2-server/pull/1146

**Comment by chalasr** [comment]
@lcobucci Thanks for the link, that should simplify things considerably indeed.
I plan to merge this one as-is as a bugfix and tag a point release for it, then open a new PR removing support for <3.4 and switching to that configuration object (it will be part of the next minor release, along with symfony 3.4 support removal)

**Comment by lcobucci** [comment]
Sounds like a plan then! Feel free to ping me on the upcoming one and thanks for your work!

**Comment by chalasr** [comment]
Thanks for your great work on lcobucci/jwt, the new API is definitely worth the hassle.

File: composer.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
     "require": {
         "php": ">=5.6",
         "ext-openssl": "*",
-        "lcobucci/jwt": "^3.2",
+        "lcobucci/jwt": "^3.2|^4.0",
```
Comment (JochenJung) [inline composer.json:42]: This change is not compatible with version 3.3.3:
Notice: Object of class DateTimeImmutable could not be converted to int in .../vendor/lcobucci/jwt/src/Builder.php line 188
You should update this to "^3.4|^4.0"

Comment (chalasr) [reply composer.json:42]: Right, the version detection logic will be fixed in #805. Thanks for the report @JochenJung

**Comment by theofidry** [comment]
@chalasr still had an issue (the one reported by https://github.com/lexik/LexikJWTAuthenticationBundle/pull/799) with lexik/jwt-authentication-bundle v2.10.4 & lcobucci/jwt 3.3.3. Upgrading to lcobucci/jwt 4.0 fixed it.

Up to you if you really want to fix it though since at least in my case updating was easier

**Pull request merged**